### DISCUSSION TITLE
EIP design systems feedback followup

### DISCUSSION THREAD

**Discussion by elainelylee** [twilio-labs/paste#1093] [Office hours] [answered]
Followup discussion from comments in invision: https://twilio.invisionapp.com/share/WXZO0CDMD49#/439932763_Cover_Sheet

**Datepicker** Currently, there isn't a datepicker in Paste so when user selects the "Custom" option in the date dropdown (Paste select), the field gets replaced with the React datepicker instead. Is there a better way to do this while we're waiting for the Paste datepickers?

**Pagination** Is there a pagination component that we can use outside of the table context?

**Keyboard accessibility** What do we need to be aware of to implement tabbing through search results in a transcript.

**Comment by SiTaggart** [comment]
Hi @elainelylee, do you have a specific question, or request for feedback on a specific piece of UI?

**Reply by elainelylee** [reply]
I've updated the description above with a few topics.

**Comment by SiTaggart** [comment, marked as answer]
Hi @elainelylee 

> Datepicker Currently, there isn't a datepicker in Paste so when user selects the "Custom" option in the date dropdown (Paste select), the field gets replaced with the React datepicker instead. Is there a better way to do this while we're waiting for the Paste datepickers?

In terms of Datepicker, it's still on our roadmap. The was to try and release an alpha by the end of the month, but it looks like that might slip into February now. Though it is most unlikely to include a date range, just a single date selection.

What I will say is though, we generally advise to never swap or replace a component with another whilst you're using it. Feel free to progressively disclose an additional field for the custom date, but please don't replace. It's a very disorientating experience for an assistive technology user and also leaves open a question of how would you get back to the fixed options if you wanted to?

In terms of waiting, the only viable solution I have to hand is just to use a plain text input that a user can enter a data into, but this might be fairly error prone. There is some discussion regarding datepickers in this thread, https://github.com/twilio-labs/paste/discussions/840

> Pagination Is there a pagination component that we can use outside of the table context?

There is no pagination component, but you are more than welcome to compose one from text, buttons and links. Here's a brief thread discussing that. https://github.com/twilio-labs/paste/discussions/938

> Keyboard accessibility What do we need to be aware of to implement tabbing through search results in a transcript.

All I can say for keyboard accessibility is just to follow the very basics. Focusable things should be made from focusable elements, most like buttons. They can be styled as you like, but underneath the html element should have a `role` of `button` and should have a `tabindex` of `0` so I can focus on it. 

That element should also respond to keyboard enter and space key presses to perform a button action, if that is the intended page level action. 

In the future we will have a button primitive to help with this, but it should be achievable right now with Paste primitives.

Aside from that, I don't think there is anything particularly special.

If on the other hand the transcript is a contenteditable html element, that poses an entirely different challenge, that I don't have a lot of advice for right now.
